zipline/utils/run_algo.py

```diff
@@ -179,6 +171,24 @@ def choose_loader(column):
         except ValueError as e:
             raise _RunAlgoError(str(e))
 
+    data = HistoricDataPortal(
```

Comment:
Would it make sense to create this only in an `else` branch below? A bunch of branches below ignore this value.